### PR TITLE
[GEOS-8475] Handle charset in contentType header when uploading style through restconfig

### PR DESCRIPTION
**Pull request by mwerlen**
Charset is ignored as it is already handled by HttpServletRequest internals.

_Contributor License Agreement already signed and sent to osgeo._

### CODE REVIEW

**Comment by mwerlen** [comment]
Hi,

You were right @ianturton, charset were not well handled by framework, my test weren't sufficient. I've added charset retrieval.

Here is my new test cases,  I've done some PUT to replace a style with accentuated signs and check if I can get it back ok. I used two files with same content, one in UTF-8 and another with ISO-8859-1. Here is the log :
```
$ file -i line-iso_8859_1.sld 
line-iso_8859_1.sld: application/xml; charset=iso-8859-1

$ file -i line-utf_8.sld 
line-utf_8.sld: application/xml; charset=utf-8

$ # Initial state
$ curl -u admin:geoserver --silent -q http://localhost:8080/geoserver/rest/styles/line.sld
<?xml version="1.0" encoding="UTF-8"?><sld:StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:sld="http://www.opengis.net/sld" xmlns:gml="http://www.opengis.net/gml" xmlns:ogc="http://www.opengis.net/ogc" version="1.0.0">
  <sld:NamedLayer>
    <sld:Name>Default Styler</sld:Name>
    <sld:UserStyle>
      <sld:Name>Default Styler</sld:Name>
      <sld:Title>Default Line</sld:Title>
      <sld:Abstract>A sample style that draws a line</sld:Abstract>
      <sld:FeatureTypeStyle>
        <sld:Name>name</sld:Name>
        <sld:Rule>
          <sld:Name>rule1</sld:Name>
          <sld:Title>Blue Line</sld:Title>
          <sld:Abstract>A solid blue line with a 1 pixel width</sld:Abstract>
          <sld:LineSymbolizer>
            <sld:Stroke>
              <sld:CssParameter name="stroke">#0000FF</sld:CssParameter>
            </sld:Stroke>
          </sld:LineSymbolizer>
        </sld:Rule>
      </sld:FeatureTypeStyle>
    </sld:UserStyle>
  </sld:NamedLayer>
</sld:StyledLayerDescriptor>

$ # UTF-8 sld PUT
$ curl -v -u admin:geoserver -X PUT http://localhost:8080/geoserver/rest/styles/line.sld -H  "accept: application/json" -H  "content-type: application/vnd.ogc.sld+xml; charset=utf-8" -d @line-utf_8.sld
* Hostname was NOT found in DNS cache
*   Trying ::1...
* Connected to localhost (::1) port 8080 (#0)
* Server auth using Basic with user 'admin'
> PUT /geoserver/rest/styles/line.sld HTTP/1.1
> Authorization: Basic YWRtaW46Z2Vvc2VydmVy
> User-Agent: curl/7.35.0
> Host: localhost:8080
> accept: application/json
> content-type: application/vnd.ogc.sld+xml; charset=utf-8
> Content-Length: 1183
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 200 OK
< Date: Thu, 14 Dec 2017 19:49:46 GMT
< X-Frame-Options: SAMEORIGIN
< Content-Length: 0
* Server Jetty(9.2.13.v20150730) is not blacklisted
< Server: Jetty(9.2.13.v20150730)
< 
* Connection #0 to host localhost left intact

$ # SLD retrieved is well encoded
$ curl -u admin:geoserver --silent -q http://localhost:8080/geoserver/rest/styles/line.sld
<?xml version="1.0" encoding="UTF-8"?><sld:StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:sld="http://www.opengis.net/sld" xmlns:gml="http://www.opengis.net/gml" xmlns:ogc="http://www.opengis.net/ogc" version="1.0.0">
  <sld:NamedLayer>
    <sld:Name>Default Styler</sld:Name>
    <sld:UserStyle>
      <sld:Name>Default Styler</sld:Name>
      <sld:Title>Default Line</sld:Title>
      <sld:Abstract>A sample style that draws a line</sld:Abstract>
      <sld:FeatureTypeStyle>
        <sld:Name>name</sld:Name>
        <sld:Rule>
          <sld:Name>rule1</sld:Name>
          <sld:Title>Blue Line</sld:Title>
          <sld:Abstract>A solid blue line with a 1 pixel test !!!! éé àà  % ^"*$¤ width</sld:Abstract>
          <sld:LineSymbolizer>
            <sld:Stroke>
              <sld:CssParameter name="stroke">#0000FF</sld:CssParameter>
            </sld:Stroke>
          </sld:LineSymbolizer>
        </sld:Rule>
      </sld:FeatureTypeStyle>
    </sld:UserStyle>
  </sld:NamedLayer>
</sld:StyledLayerDescriptor>

$ # ISO-8859-1 sld PUT
$ curl -v -u admin:geoserver -X PUT http://localhost:8080/geoserver/rest/styles/line.sld -H  "accept: application/json" -H  "content-type: application/vnd.ogc.sld+xml; charset=iso-8859-1" -d @line-iso_8859_1.sld
* Hostname was NOT found in DNS cache
*   Trying ::1...
* Connected to localhost (::1) port 8080 (#0)
* Server auth using Basic with user 'admin'
> PUT /geoserver/rest/styles/line.sld HTTP/1.1
> Authorization: Basic YWRtaW46Z2Vvc2VydmVy
> User-Agent: curl/7.35.0
> Host: localhost:8080
> accept: application/json
> content-type: application/vnd.ogc.sld+xml; charset=iso-8859-1
> Content-Length: 1183
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 200 OK
< Date: Thu, 14 Dec 2017 20:51:11 GMT
< X-Frame-Options: SAMEORIGIN
< Content-Length: 0
* Server Jetty(9.2.13.v20150730) is not blacklisted
< Server: Jetty(9.2.13.v20150730)
< 
* Connection #0 to host localhost left intact

$ # SLD is still well encoded
$ curl -u admin:geoserver --silent -q http://localhost:8080/geoserver/rest/styles/line.sld
<?xml version="1.0" encoding="UTF-8"?><sld:StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:sld="http://www.opengis.net/sld" xmlns:gml="http://www.opengis.net/gml" xmlns:ogc="http://www.opengis.net/ogc" version="1.0.0">
  <sld:NamedLayer>
    <sld:Name>Default Styler</sld:Name>
    <sld:UserStyle>
      <sld:Name>Default Styler</sld:Name>
      <sld:Title>Default Line</sld:Title>
      <sld:Abstract>A sample style that draws a line</sld:Abstract>
      <sld:FeatureTypeStyle>
        <sld:Name>name</sld:Name>
        <sld:Rule>
          <sld:Name>rule1</sld:Name>
          <sld:Title>Blue Line</sld:Title>
          <sld:Abstract>A solid blue line with a 1 pixel test !!!! éé àà  % ^"*$¤ width</sld:Abstract>
          <sld:LineSymbolizer>
            <sld:Stroke>
              <sld:CssParameter name="stroke">#0000FF</sld:CssParameter>
            </sld:Stroke>
          </sld:LineSymbolizer>
        </sld:Rule>
      </sld:FeatureTypeStyle>
    </sld:UserStyle>
  </sld:NamedLayer>
</sld:StyledLayerDescriptor>
```

File: src/main/.jdbrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+use src/main/java/
```
Comment (aaime) [inline src/main/.jdbrc]: This file seems a leftover that needs to be removed?

File: src/web/app/.jdbrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+use ../../restconfig/src/main/java/
```
Comment (aaime) [inline src/web/app/.jdbrc]: Same here, leftover file

**Comment by mwerlen** [comment]
You're right @aaime, these were leftovers.

Commits rebased, cleaned and squashed in one single commit.

**Comment by bencaradocdavies** [comment]
Close-reopen to pick up latest Travis fix on master.

**Pull request merged**